### PR TITLE
Fix incorrect pushdown in Kudu when value can be NULL.

### LINKED ISSUE #3274: Incorrect pushdown in Kudu when value can be NULL
Dear community, my friend @foxinmy found a problem we don't know if it has been Questioned@Answered.
When multiple conditions have the same column(and their logical relationship is or) then the 'is null' condition will be ignored, which lead to a problem that the data matches 'is null' condition will not be returned to the final ResultSet.
Sql is like :
`select * from profile where xwho='' or xwho is null;`
`select * from profile where distinct_id='xxx' and (xwho='a' or xwho is null)`
After debugging I think I might have found where the problem is:
in SortedRangeSet#union line 273:
image
The code below will merge conditions that has same column into one SortedRangeSet according to their local attribute lowIndexedRanges. But 'is null' condition has a 0 size lowIndexedRanges, so this condition will not be unioned to the final SortedRangeSet, and also it will not be pushed-down to the TableScanNode.
`builder.addAll(checkCompatibility(valueSet).lowIndexedRanges.values());`

### PR DESCRIPTION
**Pull request by Crossoverrr**
Fix incorrect pushdown in Kudu when value can be NULL..
image

Fixes #3274

### CODE REVIEW

**Comment by Crossoverrr** [comment]
https://github.com/prestosql/presto/issues/3274

**Comment by findepi** [comment]
@Crossoverrr LGTM. Can you try adding a test to `AbstractTestIntegrationSmokeTest`?

**Comment by Crossoverrr** [comment]
Sure! But this smoke test seems a little complex to me, I need to learn first.  :)

**Comment by Crossoverrr** [comment]
@findepi Hi, I think I have finished the smoke test.

**Comment by findepi** [comment]
@Crossoverrr would you mind checking why maven checks didn't pass?

**Comment by Crossoverrr** [comment]
> @Crossoverrr would you mind checking why maven checks didn't pass?

Sure!

**Comment by Crossoverrr** [comment]
I'm not sure why there were so many errors and I was successful in my native build.I'm using the command below
`./mvnw clean install -DskipTests`
image

**Comment by Crossoverrr** [comment]
image

**Comment by Crossoverrr** [comment]
@findepi  Hi, I think this time the building is successful!

File: presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduDistributedQueries.java [hunk 1]
```diff
@@ -74,7 +74,17 @@ public void testInsert()
     @Override
     public void testDataMappingSmokeTest(DataMappingTestSetup dataMappingTestSetup)
     {
-        // TODO Support these test once kudu connector can create tables with default partitions
```
Comment (findepi) [inline presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduDistributedQueries.java:77]: `testDataMappingSmokeTest` is a generic test that covers a few different cases.

keep this TODO here and add your test as a new test method, eg `testPredicatePushdown`

File: presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduDistributedQueries.java [hunk 2]
```diff
@@ -74,7 +74,17 @@ public void testInsert()
...
+                "id INT WITH (primary_key=true), " +
+                "col_nullable bigint with (nullable=true)" +
+                ") WITH (" +
+                " partition_by_hash_columns = ARRAY['id'], " +
+                " partition_by_hash_buckets = 2" +
+                ")");
+
+        assertUpdate("INSERT INTO test_is_null VALUES (0, 0)", 1);
+        assertUpdate("INSERT INTO test_is_null(id) VALUES (1)", 1);
+        assertQuery("SELECT count(1) FROM test_is_null WHERE col_nullable=0 or col_nullable is null", "VALUES 2");
```
Comment (findepi) [inline presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduDistributedQueries.java:87]:
```suggestion
        assertQuery("SELECT id FROM test_is_null WHERE col_nullable = 0 OR col_nullable IS NULL", "VALUES (1), (2)");
```

**Comment by Crossoverrr** [comment]
@findepi Hi, would you please help me with this building error, I can pass the test in my local debug:
image
image

**Comment by findepi** [comment]
CI failed -- https://github.com/prestosql/presto/issues/3364

**Comment by findepi** [comment]
@Crossoverrr can you please squash commits?
this will rerun CI too

let me know if you need help with this

**Comment by Crossoverrr** [comment]
@findepi Hi, I think I have finished the squash and building :)

**Comment by findepi** [comment]
Merged, thanks!

**Comment by Crossoverrr** [comment]
> Merged, thanks!

My pleasure!

**Pull request merged**